superset/hive_query.py

```diff
@@ -39,72 +16,106 @@
-                # regex for `columname` <= 1549497600
-                regex_et = "(AND)(\\s+)(`)((?:[a-z][a-z]+))(`)(\\s+)(<)(=)(\\s+)(\\d+)"
-            
-                sql_updated = re.sub(regex_st, whereClause, sql)
-                sql_updated = re.sub(regex_et, '\n', sql_updated)
-            
-                print('sql_updated ---->',sql_updated)
-            
-                return sql_updated 
-
-            else:
-                return   sql    
-
-        else:
-                return   sql  
+    # all time based  condition  should be in AND with other filters
+    if time_seq and len(time_seq) > 1:
+        where_clause = " ( " + where_clause + " ) "
 
+    return where_clause
+
+def replace_whereclause_in_org_sql(granularity, sql, where_clause, granularity_in_partitions):
+    sql_updated = sql
+    if granularity_in_partitions:
+        # regex for  `granularity` >= 1549497600 type of string
+        # any word regex (?:[a-z][a-z]+)
+        regex_st = "(`)("+granularity+")(`)(\\s+)(>)(=)(\\s+)(\\d+)"
+        # regex for `granularity` <= 1549497600
+        regex_et = "(AND)(\\s+)(`)("+granularity+")(`)(\\s+)(<)(=)(\\s+)(\\d+)"
+
+        sql_updated = re.sub(regex_st, where_clause, sql)
+        sql_updated = re.sub(regex_et, '\n', sql_updated)
     else:
-        return   sql  
\ No newline at end of file
+        sql_updated = sql.replace("WHERE", " WHERE "+ where_clause +" AND ")
+    return sql_updated
+
+def get_hive_partitions(database, datasource_name):
+    tables = list(filter(lambda tname: (tname.datasource_name == datasource_name), database.tables))
+    table = tables[0]
+    return table.hive_partitions
+
+def default_hive_query_generator(sql, query_obj, database, datasource_name):
+
+    """ schema for time based partition in table
+    {
+      "time":{
+       "year":"year",
+         "month":"month",
+        "day":"day",
+         "hour":"hour",
+        "minute":"minute"
+         }
+    }
+
+     Here we replace simply default time range based
+     where clause from sql to specific partition based  where clause
+     and returning update sql
+    """
+    if database.backend == 'hive':
+        hive_partitions = get_hive_partitions(database, datasource_name)
+        if hive_partitions:
+            hive_partitions_obj = json.loads(hive_partitions)
+            if 'time' in hive_partitions_obj:
+                time_partitions = (hive_partitions_obj['time'])
+                st = query_obj['from_dttm']
+                en = query_obj['to_dttm']
+                time_grain = query_obj['extras']['time_grain_sqla']
+                gran_seconds = get_gran_value_in_seconds(time_grain)
+                granularity = query_obj['granularity']
+                granularity_in_partitions = (granularity in time_partitions)
+                if st and en and gran_seconds:
+                    dt = datetime.today()
+                    st_seconds = dt.timestamp()
```

Comment:
can we better name this to suggest we are uisng it for logging query formation time. I am bit confused in mixing it with `st`

Reply:
updated